Source/WebCore/style/AnchorPositionEvaluator.cpp

```diff
@@ -725,76 +732,78 @@ AnchorElements AnchorPositionEvaluator::findAnchorsForAnchorPositionedElement(co
     return anchorElements;
 }
 
-void AnchorPositionEvaluator::updateAnchorPositioningStatesAfterInterleavedLayout(const Document& document)
+void AnchorPositionEvaluator::updateAnchorPositioningStatesAfterInterleavedLayout(Document& document)
 {
-    if (document.styleScope().anchorPositionedStates().isEmptyIgnoringNullReferences())
-        return;
-
-    auto anchorsForAnchorName = collectAnchorsForAnchorName(document);
-
-    for (auto elementAndState : document.styleScope().anchorPositionedStates()) {
-        auto& state = *elementAndState.value;
-        if (state.stage == AnchorPositionResolutionStage::Initial) {
-            Ref element { elementAndState.key };
-            if (element->renderer())
-                state.anchorElements = findAnchorsForAnchorPositionedElement(element, state.anchorNames, anchorsForAnchorName);
-            state.stage = AnchorPositionResolutionStage::FoundAnchors;
-            continue;
+    visitAllAnchorPositionedStates(document, [&document](auto& states) {
```

Comment:
Can you explain what moving the anchor positioned element map to individual style scopes achieves concretely?

Reply:
It basically introduces a true isolation of anchoring-related data. This way it's possible e.g. to have same anchor names in unrelated shadow trees and not run into conflicts.

In my mind, such separation was the easiest first step to have basic shadow tree isolation working with our current implementation without introducing any substantial refactoring or implementing support for shadow parts at the same time.